$DragonFly: src/lib/libarchive/patches/Attic/archive_string_sprintf.c.patch,v 1.1 2004/11/10 18:02:24 joerg Exp $

--- archive_string_sprintf.c.orig	2004-11-09 23:59:26.000000000 +0100
+++ archive_string_sprintf.c	2004-11-10 00:40:12.000000000 +0100
@@ -47,13 +47,21 @@
     va_list ap)
 {
 	size_t l;
+#ifdef HAVE_VA_COPY
 	va_list ap1;
+#endif
 
 	if (fmt == NULL) {
 		as->s[0] = 0;
 		return;
 	}
 
+#ifdef HAVE_VASPRINTF
+	archive_string_free(as);
+	l = vasprintf(&as->s, fmt, ap);
+	as->buffer_length = l + 1;
+	as->length = l;
+#elif defined(HAVE_VA_COPY)
 	va_copy(ap1, ap);
 	l = vsnprintf(as->s, as->buffer_length, fmt, ap);
 	/* If output is bigger than the buffer, resize and try again. */
@@ -63,4 +71,7 @@
 	}
 	as->length = l;
 	va_end(ap1);
+#else
+#error "Either vasprintf or va_copy are needed"
+#endif
 }
